dbus/src/message/matchrule.rs

```diff
@@ -167,4 +167,300 @@ impl<'a> MatchRule<'a> {
         self.msg_type = Some(ty);
         self
     }
+
+    #[cfg(feature = "parser")]
+    /// Tries parsing a MatchRule from a String. Please note however that not all features supported
+    /// by DBus are supported by dbus-rs (yet). args and destinations are not supported yet.
+    pub fn parse(text: &'a str) -> Result<Self, parser::Error> {
+        parser::Parser::new(text)?.parse()
+    }
+}
+
+
+#[cfg(feature = "parser")]
+pub mod parser {
```

Comment:
How about moving this module to its own file?

Reply:
To be honest when I started writing the parser I was unsure how much work it would actually entail. I'll move it in the next commit.